src/userscripts.ts

```diff
@@ -96,6 +103,28 @@ class Userscript implements IUserscriptInstance {
 			if (typeof exported !== 'undefined') {
 				// more stuff to be added here later
 				if ('unload' in exported) this.unload = exported.unload;
+				if ('settings' in exported) {
+					this.settings = exported.settings
+					strippedConsole.log(`Re-defined userscript settings.`)
+				}
+			}
+
+			if (existsSync(this.settingsPath) && Object.keys(this.settings).length > 0) {
+				try {
+					var settingsJSON = JSON.parse(readFileSync(this.settingsPath, 'utf-8'));
+					Object.keys(settingsJSON).forEach(settingKey => {
```

Comment:
this is the same validation chain as above, please extract it to a function that gets re-used